sentinelsat/sentinel.py

```diff
@@ -145,13 +144,13 @@ def query(self, area, initial_date=None, end_date=datetime.now(), **keywords):
         and any other search keywords accepted by the SciHub API.
         """
         query = self.format_query(area, initial_date, end_date, **keywords)
-        self.load_query(query)
-        return self.products
+        return self.load_query(query)
 
-    def load_query(self, query, start_row=0):
+    def load_query(self, query, start_row=0, products=[]):
         """Do a full-text query on the SciHub API using the format specified in
            https://scihub.copernicus.eu/twiki/do/view/SciHubUserGuide/3FullTextSearch
         """
+        output = products
```

Comment:
I tried the following diff, but the failures are the same

```
diff --git a/sentinelsat/sentinel.py b/sentinelsat/sentinel.py
index b15f263..baa83fa 100644
--- a/sentinelsat/sentinel.py
+++ b/sentinelsat/sentinel.py
@@ -150,7 +150,6 @@ class SentinelAPI(object):
         """Do a full-text query on the SciHub API using the format specified in
            https://scihub.copernicus.eu/twiki/do/view/SciHubUserGuide/3FullTextSearch
         """
-        output = products
         # store last query (for testing)
         self.last_query = query
 
@@ -168,14 +167,14 @@ class SentinelAPI(object):
         total_results = 0
         try:
             json_feed = content.json()['feed']
-            products = json_feed['entry']
+            entries = json_feed['entry']
             # this verification is necessary because if the query returns only
             # one product, self.products will be a dict not a list
-            if type(products) == dict:
-                products = [products]
+            if type(entries) == dict:
+                entries = [entries]
 
             # append to products
-            output += products
+            products += entries
 
             # get total number of returned results
             total_results = int(json_feed['opensearch:totalResults'])
@@ -191,8 +190,8 @@ class SentinelAPI(object):
 
         # repeat query until all results have been loaded
         if total_results > self.page_size + start_row - 1:
-            self.load_query(query, start_row=(start_row + self.page_size), products=output)
-        return output
+            self.load_query(query, start_row=(start_row + self.page_size), products=products)
+        return products
 
     @staticmethod
     def format_query(area, initial_date=None, end_date=datetime.now(), **keywords):
```